### PR TITLE
Add update to runtimes considerations

### PR DESCRIPTION
**Pull request by lzdanski**
- https://github.com/astronomer/docs/issues/2243
- Added content from slack conversation to Runtimes considerations
- Also updated style conventions between considerations for past Astro Runtimes

### CODE REVIEW

File: astro/upgrade-runtime.md [hunk 1]
```diff
@@ -42,6 +42,13 @@ Astro Runtime 8 includes fewer default dependencies than previous versions. Spec
 
 If your DAGs depend on any of these provider packages, add the provider packages to your Astro project `requirements.txt` file before upgrading. You can also [pin specific provider package versions](#optional-pin-provider-package-versions) to ensure that none of your provider packages change after upgrading.
 
+There is also an incompatibility between Astro Runtime 8.0.0 and the following provider packages:
+
+- `apache-airflow-providers-cncf-kubernetes==6.1.0`
+- `apache-airflow-providers-google==10.0.0`
+
+Runtime 8.0.0 includes both of these packages as built-in components. A new function included in this version, `get_xcom_sidecar_container_resources`, is used in `KubernetesHook`. But, Google because uses `GKEPodHook`, which doesn’t have this function, it breaks `GKEStartPodOperator`.
+
```
Comment (abhishekbhakat) [inline astro/upgrade-runtime.md:51]: @lzdanski please add a suggestion to pin `apache-airflow-providers-cncf-kubernetes==5.2.2` in `requirements.txt` as a fix.

**Review by abhishekbhakat — APPROVED** [review]
LGTM

File: astro/upgrade-runtime.md [hunk 2]
```diff
@@ -42,6 +42,15 @@ Astro Runtime 8 includes fewer default dependencies than previous versions. Spec
...
 
+There is also an incompatibility between Astro Runtime 8.0.0 and the following provider packages:
+
+- `apache-airflow-providers-cncf-kubernetes==6.1.0`
+- `apache-airflow-providers-google==10.0.0`
+
+That can be resolved by pinning `apache-airflow-providers-cncf-kubernetes==5.2.2` in your `requirements.txt`file.
+
+This incompatibility occurs because Runtime 8.0.0 includes both of these packages as built-in components. A new function included in this version, `get_xcom_sidecar_container_resources`, is used in `KubernetesHook`. But, Google because uses `GKEPodHook`, which doesn’t have this function, it breaks `GKEStartPodOperator`. 
+
```
Comment (jwitz) [inline astro/upgrade-runtime.md:53]:
```suggestion

##### Upgrade to Python 3.10

Astro Runtime 8 uses Python 3.10. If you use provider packages that don't yet support Python 3.10, use one of the following options to stay on Python 3.9:

- Run your tasks using the KubernetesPodOperator or PythonVirtualenvOperator. You can configure the environment that these tasks run in to use Python 3.9.
- Use the [`astronomer-provider-venv`](https://github.com/astronomer/astro-provider-venv) to configure a custom virtual environment that you can apply to individual tasks.
```

**Review by jwitz — APPROVED** [review]
LGTM, feel free to play with my wording but I also added a suggestion/ change to include the Python 3.10 runtime considerations here

**Pull request merged**